service/dap/server.go

```diff
@@ -385,8 +384,8 @@ func (s *Server) recoverPanic(request dap.Message) {
 
 func (s *Server) handleRequest(request dap.Message) {
 	defer s.recoverPanic(request)
-	s.resumeMu.Lock()
-	defer s.resumeMu.Unlock()
+	s.continueMu.Lock()
```

Comment:
That example was from when I thought that the UI would assume the halt succeeded when it receives a response, which it does not. Because of this the current code does not actually address this.

Though maybe it needs to depending on if Halt while paused clears the internal breakpoints.